docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,127 @@
+        matchLabels:
+            name: some-app
+```
+
+The above YAML will have the Application depend on any other Application that matches the given label selector. In this case, any application carrying a label `name` with the value of `some-app`.
+
+### Matching on name patterns
+
+Dependencies can be matched on the name of Applications by specifying the `namePattern` selector. The `namePattern` selector takes a list of Application name patterns to match against the Applications' `.metadata.name` property. Specifying multiple patterns 
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - namePattern:
+      - some-app
+      - other-app
+```
+
+The above YAML will have the Application depend on the Applications `some-app` and `other-app`. 
+
+The `namePattern` matcher supports shell-style glob matching, so the following example would match all Applications whose name is starting with `stage1-`:
+
+```yaml
+spec:
+  dependsOn:
+    selectors:
+    - namePattern:
+      - stage1-*
+```
+
+Regular expressions are not supported.
+
+## Configuring behaviour of dependency resolution
+
+There are several options to configure the behaviour of dependency resolution. Please note that these settings are not inherited to dependencies, but must be configured for each dependant uniquely.
+### Delayed start of automated sync
+
+In certain scenarios, you may want to delay the automatic sync process when there are dependencies configured between applications that are created at the same time (i.e. synced through an app of apps pattern from Git). This may allow the synchronization of all applications to succeed before the dependency tree is resolved and the sync is actually started in the right order.
+
+To configure the sync delay, specify the number of seconds to delay in the field `syncDelay` as follows:
+
+```yaml
+spec:
+  dependsOn:
+    syncDelay: 5
+```
+
+Please note that any delay you configure here will delay _any_ automatic sync, regardless of whether the Application just has been created or existed for a while.
+
+### Block on empty dependencies
+
+It is possible to block syncing when dependency selectors are configured, but no dependency exists yet. This may be useful in scenarios where you want to prevent an Application to be synced unless another specific Application exists.
+
+To configure blocked syncs on empty dependencies, use the field `blockOnEmpty` as follows:
+
+```yaml
+spec:
+  dependsOn:
+    blockOnEmpty: true
+```
+
+Note that when you set `blockOnEmpty` to true, and sync this app through the CLI, the `argocd app sync` command will only return when the application starts syncing, i.e. when it has finished waiting. If you need the CLI to return immediately, use `argocd app sync --async`.
+
+### Configure a wait timeout
+
+You can configure the maximum time that Argo CD should wait for dependencies to be resolved and the sync to start. The timeout also applies to the wait spent in blocking for empty dependencies.
```

Comment:
would this change the behavior of `argocd app sync` (mentioned above)?

Reply:
Good question actually, I haven't tested behaviour with the CLI yet. I'll follow up on this.